lib/src/main/kotlin/com/swmansion/starknet/data/serializers/JsonRpcTransactionPayloadSerializer.kt

```diff
@@ -0,0 +1,53 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.data.types.transactions.*
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+@OptIn(ExperimentalSerializationApi::class)
+@Serializer(forClass = TransactionPayload::class)
+object JsonRpcTransactionPayloadSerializer : KSerializer<TransactionPayload> {
```

Comment:
Not sure about the name, could leave it as `JsonRpcTransactionPayloadPolymorphicSerializer`, but that implies inheriting from `JsonContentPolymorphicSerializer<T>` from what I understand, which is not the case here.